hibernate-core/src/main/java/org/hibernate/mapping/DenormalizedTable.java

```diff
@@ -83,6 +84,20 @@ public void createForeignKeys(MetadataBuildingContext context) {
 		}
 	}
 
+	private ForeignKey createNormalizedForeignKey(ForeignKey includedTableFk) {
+		ForeignKey normalizedForeignKey = new ForeignKey(this);
+
+		normalizedForeignKey.setReferencedEntityName(includedTableFk.getReferencedEntityName());
+		normalizedForeignKey.setKeyDefinition(includedTableFk.getKeyDefinition());
+		normalizedForeignKey.setReferencedTable(includedTableFk.getReferencedTable());
+		normalizedForeignKey.addReferencedColumns(includedTableFk.getReferencedColumns());
+		for (Column keyColumn : includedTableFk.getColumns()) {
+			normalizedForeignKey.addColumn(keyColumn);
+		}
+
+		return normalizedForeignKey;
+	}
```

Comment:
This might be better expressed as a copy constructor, as it will be less likely to forget to update this logic if e.g. a new field is ever added to the `ForeignKey` class.